pkg/services/apiserver/storage/entity/storage.go

```diff
@@ -511,9 +511,14 @@ func (s *Storage) GuaranteedUpdate(
 		Subresource: requestInfo.Subresource,
 	}
 
-	err := s.Get(ctx, k.String(), storage.GetOptions{}, destination)
-	if err != nil {
-		return err
+	getErr := s.Get(ctx, k.String(), storage.GetOptions{}, destination)
```

Comment:
could this be simplified by passing [ignoreNotFound to GetOptions](https://github.com/kubernetes/kubernetes/blob/227c2e7c2b2c05a9c8b2885460e28e4da25cf558/staging/src/k8s.io/apiserver/pkg/storage/interfaces.go#L260)?

Reply:
good call, will check that out

Reply:
I checked and we need logic to handle the not found case either way, so it's actually a bit clearer this way vs having to check whether destination was populated or not.